engine/src/main/java/org/hibernate/search/engine/search/highlight/dsl/QueryHighlightConfigurationFinalStep.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.highlight.dsl;
+
+import org.hibernate.search.engine.search.highlight.HighlightConfiguration;
+
+/**
+ * TODO
+ */
+public interface QueryHighlightConfigurationFinalStep {
+
+	HighlightConfiguration toConfiguration();
```

Comment:
That's actually a nice idea. Though, to make the most of it, you'd have to expose the highlight configuration factory all the way up to `SearchScope`, so there's a bit more work involved.

Also, to be consistent, I'd rename `HighlightConfiguration` to `SearchHighlightConfiguration`, or (preferably, if we go with my suggestion to use `.highlighter()` in the DSL, `SearchHighlighter`.